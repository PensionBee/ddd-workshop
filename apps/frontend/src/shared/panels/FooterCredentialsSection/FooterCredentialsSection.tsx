import { GridContainer, GridItem, GridRow } from "@shared/components/Grid";
import { FooterSystemIcon } from "@shared/components/Svg";
import { DisclaimerL } from "@shared/components/Typography";
import { Link } from "@shared/components/Link";

export const FooterCredentialsSection: React.FC = (props) => {
  return (
    <GridContainer
      data-rc="FooterCredentialsSection"
      pt={{ xs: 40, lg: 60 }}
      pb={{ xs: 40, lg: 60 }}
      {...props}
    >
      <GridRow rowGap={{ xs: 40 }}>
        <GridItem span={{ lg: 4 }} align="center">
          <FooterSystemIconLink
            icon="FscsLogo"
            href="https://www.fscs.org.uk/what-we-cover/"
            target="_blank"
          />
        </GridItem>
        <GridItem span={{ lg: 4 }} align="center">
          <FooterSystemIconLink
            icon="TrustPilot"
            href="https://uk.trustpilot.com/review/pensionbee.com?utm_medium=trustboxes&utm_source=MicroTrustScore"
            target="_blank"
            text="Excellent 4.6 out of 5"
          />
        </GridItem>
        <GridItem span={{ lg: 4 }} align="center">
          <FooterSystemIconLink
            icon="InternetCrystalMark"
            href="https://www.plainenglish.co.uk/accreditation.html"
            target="_blank"
          />
        </GridItem>
      </GridRow>
    </GridContainer>
  );
};

type FooterSystemIconLinkProps = React.ComponentProps<
  typeof FooterSystemIcon
> & {
  href: React.ComponentProps<typeof Link>["href"];
  target: React.ComponentProps<typeof Link>["target"];
  text?: string;
};

const FooterSystemIconLink: React.FC<FooterSystemIconLinkProps> = ({
  icon,
  text,
  ...props
}) => {
  return (
    <Link {...props}>
      <FooterSystemIcon icon={icon} />
      <DisclaimerL as="p" align="center">
        {text}
      </DisclaimerL>
    </Link>
  );
};
