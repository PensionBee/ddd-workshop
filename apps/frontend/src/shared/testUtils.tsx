import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import type { Breakpoints } from "./styling/breakpoints";
import { BREAKPOINTS } from "./styling/breakpoints";
import { theme } from "./styling/theme";
import { vi } from "vitest"

type ReactTestingLibraryProviderProps = {
  children: React.ReactNode;
};

// eslint-disable-next-line react-refresh/only-export-components
const ReactTestingLibraryProviders: React.FC<
  ReactTestingLibraryProviderProps
> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const customRender = (ui: React.ReactElement, options?: any) => {
  return render(ui, { wrapper: ReactTestingLibraryProviders, ...options });
};

// re-export everything
// eslint-disable-next-line react-refresh/only-export-components
export * from "@testing-library/react";

// override render method
export { customRender as render };

export const sleep = (delay: number) => {
  return new Promise((resolve) => setTimeout(resolve, delay));
};

export const mockMediaMatches = (breakpoint: Breakpoints) => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: (query: string) => {
      const breakpointValue = BREAKPOINTS[breakpoint];
      const queryValue = parseInt(query.replace(/^\D+/g, ""));
      const breakpointValues = [...Object.values(BREAKPOINTS)];
      if (!breakpointValues.includes(queryValue)) {
        throw new Error(
          `Invalid breakpoint value: ${queryValue}, this likely means there has been a change to the way that breakpoints are setup.`
        );
      }
      return {
        matches: queryValue <= breakpointValue,
        addEventListener: vi.fn(),
        removeEventListener: vi.fn(),
      };
    },
  });
};

export const routerRender = (ui: React.ReactNode, pathname = "/") => {
  return customRender(
    <MemoryRouter initialEntries={[{ pathname }]}>{ui}</MemoryRouter>
  );
};
