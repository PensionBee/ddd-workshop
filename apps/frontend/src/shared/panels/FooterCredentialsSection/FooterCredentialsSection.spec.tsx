import { render, vi } from "@shared/testUtils";
import { FooterCredentialsSection } from "./FooterCredentialsSection";

describe("FooterCredentialsSection", () => {
  const TEST_ID = "FooterCredentialsSection";
  it("should render", () => {
    const { getByTestId, asFragment } = render(
      <FooterCredentialsSection data-testid={TEST_ID} />
    );
    expect(getByTestId(TEST_ID)).toBeInTheDocument();
    expect(asFragment).toMatchSnapshot();
  });
});
