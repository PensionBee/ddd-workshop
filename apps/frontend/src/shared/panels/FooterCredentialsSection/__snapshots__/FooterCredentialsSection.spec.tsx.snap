// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FooterCredentialsSection > should render 1`] = `[Function]`;

exports[`FooterCredentialsSection should render 1`] = `[Function]`;
